File: BoSeonPark/투 포인터/2470.py
import sys
input = sys.stdin.readline

n = int(input())
acid = list(map(int, input().split()))
acid.sort()
l, r = 0, n-1
a, b, c = 0, 0, 2e+9+1

while l != r:
    temp = acid[l] + acid[r]
    if abs(temp) < c:
        a, b, c = acid[l], acid[r], abs(temp)
    
    if temp < 0:
        l += 1
    else:
        r -= 1

print(a, b)

File: 김하영/two pointer/2470.js
const fs = require("fs");

const filePath = process.platform === "linux" ? "/dev/stdin" : "input.txt";

const input = fs.readFileSync(filePath).toString().trim().split("\n");

const n = +input[0];
const nums = input[1].split(" ").map(Number);

const solution = (n, nums) => {
  nums.sort((a, b) => a - b);
  let start = 0;
  let end = n - 1;
  let sum = Infinity;
  let a = 0;
  let b = 0;
  let temp = Infinity;

  while (start < end) {
    sum = nums[start] + nums[end];
    if (sum < 0) {
      if (Math.abs(sum) < Math.abs(temp)) {
        temp = sum;
        a = nums[start];
        b = nums[end];
      }
      start++;
    } else {
      if (Math.abs(sum) < Math.abs(temp)) {
        temp = sum;
        a = nums[start];
        b = nums[end];
      }
      end--;
    }
  }

  if (nums.indexOf(a) > nums.indexOf(b)) {
    return `${b} ${a}`;
  } else {
    return `${a} ${b}`;
  }
};
console.log(solution(n, nums));
